src/net_processing.cpp

```diff
@@ -909,11 +907,16 @@ void PeerManager::FinalizeNode(NodeId nodeid, bool& fUpdateConnectionTime) {
     LogPrint(BCLog::NET, "Cleared nodestate for peer=%d\n", nodeid);
 }
 
-PeerRef PeerManager::GetPeerRef(NodeId id)
+PeerRef PeerManager::GetPeerRef(NodeId id, bool erase)
```

Comment:
This is (mostly) a style preference, but I'm not a fan of adding a boolean option to substantially change the functionality of a function as opposed to just creating two functions. GetPeerRef() is logically a read-only operation.

It's also easier easy to slip up and make mistakes like this one: https://github.com/bitcoin/bitcoin/pull/19910#discussion_r499754354 ;)

Reply:
You got me. I've made it a separate function.

> GetPeerRef() is logically a read-only operation.

Good point. I've made it `const` (and made the mutex `mutable`)